redfish-core/lib/log_services.hpp

```diff
@@ -5368,4 +5419,1084 @@ inline void requestRoutesPostCodesEntry(App& app)
+            for (const auto& ancestorIface : ancestorsIfaces)
+            {
+                bool foundAncestor = false;
+                for (const auto& obj : ancestors)
+                {
+                    for (const auto& service : obj.second)
+                    {
+                        for (const auto& interface : service.second)
+                        {
+                            if (interface == ancestorIface.first)
+                            {
+                                foundAncestor = true;
+                                redfishUri.replace(
+                                    ancestorIface.second, uriIdPattern.length(),
+                                    getIsolatedHwItemId(
+                                        sdbusplus::message::object_path(
+                                            obj.first)));
+                                if (isChassisAssemblyUri &&
+                                    interface ==
+                                        "xyz.openbmc_project.Inventory."
+                                        "Item.Chassis")
+                                {
+                                    assemblyParent = std::make_tuple(
+                                        service.first,
+                                        sdbusplus::message::object_path(
+                                            obj.first),
+                                        interface);
+                                }
+                                break;
+                            }
+                        }
+                        if (foundAncestor)
+                        {
+                            break;
+                        }
+                    }
+                    if (foundAncestor)
+                    {
+                        break;
+                    }
+                }
+
+                if (!foundAncestor)
+                {
+                    BMCWEB_LOG_ERROR
+                        << "Failed to fill Links:OriginOfCondition "
+                        << "because unable to get parent DBus path "
+                        << "for the identified parent interface : "
+                        << ancestorIface.first
+                        << " of the given DBus object path: "
+                        << dbusObjPath.str;
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+            }
+
+            if (entryJsonIdx > 0)
+            {
+                asyncResp->res.jsonValue["Members"][entryJsonIdx - 1]["Links"]
+                                        ["OriginOfCondition"] = {
+                    {"@odata.id", redfishUri}};
+
+                if (isChassisAssemblyUri)
+                {
+                    auto uriPropPath = "/Members"_json_pointer;
+                    uriPropPath /= entryJsonIdx - 1;
+                    uriPropPath /= "Links";
+                    uriPropPath /= "OriginOfCondition";
+
```

Comment:
This needs odata.id like we have today: 
https://github.com/ibm-openbmc/bmcweb/blob/1030/redfish-core/lib/log_services.hpp#L4799

Reply:
Thank you! Added to the refactor JIRA ticket https://jsw.ibm.com/browse/PFEBMC-620 and will be addressed as part of it

Reply:
we should address this here, otherwise we have a validator error

Reply:
Addressed it. Here is the link to redfish validator results.
https://sys-openbmc-dev-jenkins.swg-devops.com/job/OpenBMC-Dev/job/test/job/common/job/Redfish-Service-Validator/18118/artifact/logs/ConformanceHtmlLog_02_20_2023_133127.html